# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/akonadi/hotfix-cross-protocolgen.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- a/src/private/CMakeLists.txt
+++ b/src/private/CMakeLists.txt
@@ -11,14 +11,22 @@
     )
 endif()

+if(CMAKE_CROSSCOMPILING AND PROTOCOLGEN_EXECUTABLE)
+    set(protocolgen_cmd ${PROTOCOLGEN_EXECUTABLE})
+    set(protocolgen_dep)
+else()
+    set(protocolgen_cmd protocolgen)
+    set(protocolgen_dep protocolgen)
+endif()
+
 add_custom_command(
     OUTPUT
         ${CMAKE_CURRENT_BINARY_DIR}/protocol_gen.cpp
         ${CMAKE_CURRENT_BINARY_DIR}/protocol_gen.h
     COMMAND
-        protocolgen ${CMAKE_CURRENT_SOURCE_DIR}/protocol.xml
+        ${protocolgen_cmd} ${CMAKE_CURRENT_SOURCE_DIR}/protocol.xml
     DEPENDS
-        protocolgen
+        ${protocolgen_dep}
         ${CMAKE_CURRENT_SOURCE_DIR}/protocol.xml
     COMMENT "Generating Protocol implementation"
 )
